lower_acl_than_parent not safe

### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

Applying `lower_acl_than_parent` to provided sample code breaks it.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
Linting Swift files in current working directory
Linting 'Example.swift' (1/1)
/path/Example.swift:3:5: warning: Lower ACL than Parent Violation: Ensure declarations have a lower access control level than their enclosing parent (lower_acl_than_parent)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.51.0
* Installation method used (Homebrew, CocoaPods, building from source, etc)? homebrew
* Paste your configuration file:

```yml
only_rules:
- lower_acl_than_parent
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No.
* Which Xcode version are you using (check `xcodebuild -version`)? Xcode 14.3 Build version 14E222b
* Do you have a sample that shows the issue?

```swift
struct Example { 
    let content: String
    public init(content: String) {}
}
```
is corrected to:
```swift
struct Example { 
    let content: Stringinit(content: String) {}
}
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -25,7 +25,9 @@
 
 #### Bug Fixes
 
-* None.
+* Fix `lower_acl_than_parent` rule rewriter by preserving leading whitespace.  
+  [SimplyDanny](https://github.com/SimplyDanny)
+  [#4860](https://github.com/realm/SwiftLint/issues/4860)
 
 ## 0.51.0: bzllint
 

--- a/Source/SwiftLintFramework/Rules/Lint/LowerACLThanParentRule.swift
+++ b/Source/SwiftLintFramework/Rules/Lint/LowerACLThanParentRule.swift
@@ -68,7 +68,17 @@ struct LowerACLThanParentRule: OptInRule, ConfigurationProviderRule, SwiftSyntax
             Example("class Foo { ↓public func bar() {} }"):
                 Example("class Foo { func bar() {} }"),
             Example("actor Foo { ↓public func bar() {} }"):
-                Example("actor Foo { func bar() {} }")
+                Example("actor Foo { func bar() {} }"),
+            Example("""
+                struct Foo {
+                    ↓public func bar() {}
+                }
+                """):
+                Example("""
+                struct Foo {
+                    func bar() {}
+                }
+                """)
         ]
     )
 
@@ -120,7 +130,10 @@ private extension LowerACLThanParentRule {
                     trailingTrivia: .space
                 )
             } else {
-                newNode = DeclModifierSyntax(name: .keyword(.internal, presence: .missing))
+                newNode = DeclModifierSyntax(
+                    leadingTrivia: node.leadingTrivia ?? .zero,
+                    name: .identifier("")
+                )
             }
 
             return super.visit(newNode)

